### PR TITLE
Code reorganization. Fixes #19, fixes #21, fixes #23.

### LINKED ISSUE #19: Organize OCL-related code into submodules
- Put at least the following into their own submodules: 
  - Put all OpenCL related modules into their own 'ocl' submodule and get rid of the 'OCL' in the file names: AFMulatorOCL, HighLevelOCL, RelaxOpenCL, fieldOCL, oclUtils.
   - Put all machine learning related modules into their own 'ml' submodule: AuxMap, CorrectionLoop, Corrector, GeneratorOCL.

### LINKED ISSUE #21: Rename GUILJ to something simpler like 'ppm' or 'ppm-gui'.

### LINKED ISSUE #23: Rename GeneratorOCL_Simple2 -> Generator and delete other versions.
Rename GeneratorOCL_Simple2 -> Generator and delete other versions. I'm probably the only one using this so should not be a problem.

### PR DESCRIPTION
**Pull request by NikoOinonen**
fixes #19
fixes #21
fixes #23

Moved AFMulatorOCL_Simple, HighLevelOCL, RelaxOpenCL, fieldOCL and oclUtils to a submodule called ocl and simplified the names of the modules.

Moved AuxMap, CorrectionLoop, Corrector and GeneratorOCL_Simple2 to a submodule called ml and renamed GeneratorOCL_Simple2 to just Generator.

Removed other old versions of GeneratorOCL and AFMulatorOCL.

Renamed GUILJ.py to ppm-gui.

### CODE REVIEW

File: examples/Generator/test_trainer.py
Comment (yakutovicha) [inline examples/Generator/test_trainer.py:29]: I think `lvec` is missing, no?

**Review by yakutovicha — CHANGES_REQUESTED** [review]
Thanks, @NikoOinonen. Just one minor detail. The rest is good.

File: examples/Generator/test_trainer.py
Comment (NikoOinonen) [reply examples/Generator/test_trainer.py:29]: `lvec` is now an optional argument that gets automatically inferred from the scan window if it's not specified. The behaviour changed since the test was last updated.

Comment (yakutovicha) [reply examples/Generator/test_trainer.py:29]: ok, all good then.

**Comment by yakutovicha** [comment]
@NikoOinonen please resolve the merging conflicts, so we can merge the PR.

Thanks!

**Comment by ProkopHapala** [comment]
Perhaps I need some instruction, if I should review it, what exactly should I do? Simple checkout on my machine in separate copy of repo? Or can I look on it online on github?

File: tests/FARFF/corrector.sh
Comment (ProkopHapala) [inline tests/FARFF/corrector.sh:17]: Did you tried to run correction loop ?

File: pyProbeParticle/ocl/field.py
Comment (ProkopHapala) [inline pyProbeParticle/ocl/field.py:15]: does it mean it can run also without reikna? if we run just point-charge electrostatics and don't need FFT ?

File: examples/PTCDA_Hartree/run_gpu_easy.py
Comment (ProkopHapala) [inline examples/PTCDA_Hartree/run_gpu_easy.py:8]: "easy" means without Hartree potential (with point charges) right ?

**Review by ProkopHapala — COMMENTED** [review]
I did not try to run anything, but from what I see I agree. I left several questions/comments. Hope they are visible

File: examples/Generator/test_trainer.py
Comment (NikoOinonen) [reply examples/Generator/test_trainer.py:30]: Well, this is just some example script and the dict was not being used for anything else than initializing the AFMulator object, so I thought to simplify it. There is no reason why you could not still use a dict if you want to.

File: tests/FARFF/corrector.sh
Comment (NikoOinonen) [reply tests/FARFF/corrector.sh:17]: I tried running the script at examples/CorrectionLoopGraphene/test_CorrectionLoop.py and at least it completes successfully.

File: pyProbeParticle/ocl/field.py
Comment (NikoOinonen) [reply pyProbeParticle/ocl/field.py:15]: Correct. I made it this way because I did not want to break some scripts that I had running in environments without reikna. Eventually when we get this to PyPI, we will probably add reikna as a dependency, so could then get rid of this try-except.

File: examples/PTCDA_Hartree/run_gpu_easy.py
Comment (NikoOinonen) [reply examples/PTCDA_Hartree/run_gpu_easy.py:8]: I called it easy, because it uses this `quick_afm` convenience function that I made so that you can quickly get some AFM images just by giving an input file and some physical parameters without needing to care about the OCL stuff. It works both with point charges and Hartree.

**Comment by NikoOinonen** [comment]
> Perhaps I need some instruction, if I should review it, what exactly should I do? Simple checkout on my machine in separate copy of repo? Or can I look on it online on github?

You can do whatever you want to make sure that you are happy with the changes. If simply looking at the changes on Github is enough for you, then that's good too.

Since @yakutovicha already approved, I could merge it, but I'll wait until @ProkopHapala is okay with the changes before merging.

**Comment by yakutovicha** [comment]
@ProkopHapala, do you know how to approve?

**Comment by ProkopHapala** [comment]
Not sure If I did the right think. I perhaps merged it already. Approve is something else ?

**Comment by NikoOinonen** [comment]
@ProkopHapala It got merged already. I might have been better if it was squash and merge, but it's fine. It does not show an approval from you for me, though.

**Comment by yakutovicha** [comment]
Yeah, Prokop just merged it directly.

**Pull request merged**
